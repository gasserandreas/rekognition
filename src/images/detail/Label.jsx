import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Colors } from '../../styles';

const handleLabelHover = (props) => props.isClickable
  ? `
    color: ${Colors.Blue.Default};
    border-color: ${Colors.Blue.Default};

    &:hover {
      cursor: pointer;
      background-color: ${Colors.Blue.Default};
      color: ${Colors.ColorsPalette.White};
    }
  ` : '';

const StyledLabel = styled.div`
  border: 1.5px solid ${Colors.Neutrals.Mid};
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;

  span {
    font-size: 0.75rem;
  }

  ${(props) => handleLabelHover(props)}
`;

const Label = (props) => {
  const { label, ...rest } = props;
  const { name, confidence } = label;

  const confidenceStr = String(confidence).substring(0,5);

  return (
    <StyledLabel
      {...rest}
    >
      {name} { confidenceStr && <span>({confidenceStr})</span>}
    </StyledLabel>
  );
};

Label.propTypes = {
  label: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    confidence: PropTypes.number.isRequired,
    parents: PropTypes.arrayOf(PropTypes.string),
    instances: PropTypes.arrayOf(PropTypes.shape({
      height: PropTypes.number.isRequired,
      width: PropTypes.number.isRequired,
      left: PropTypes.number.isRequired,
      top: PropTypes.number.isRequired,
    })),
  }).isRequired,
  onClick: PropTypes.func,
  isClickable: PropTypes.bool,
};

Label.defaultProps = {
  onClick: null,
  isClickable: false,
};

export default Label;
