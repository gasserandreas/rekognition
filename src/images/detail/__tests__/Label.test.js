import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Label from '../Label';

describe('Faces test suite', () => {
  const mockedLabel = {
    id: 'f7ba4488-4db3-484d-9af9-37f1a7a2aeb2',
    name: 'name',
    confidence: 100,
    value: 'value',
    parents: [],
    instances: [{
      height: 100,
      width: 90,
      left: 80,
      top: 70,
    }],
  };

  const initialProps = {
    label: mockedLabel,
    selected: false,
    isClickable: true,
    onClick: jest.fn(),
  };

  const getLabel = (props) => mount(
    <Label
      {...initialProps}
      {...props}
    />
  );

  afterEach(() => {
    initialProps.onClick.mockClear();
  });

  it('should render label', () => {
    const wrapper = getLabel();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('#jestConfidence').exists()).toBeTruthy();
  });

  it('should not render consistence if undefined', () => {
    const wrapper = getLabel({
      ...initialProps,
      label: {
        ...initialProps.label,
        confidence: undefined,
      },
    });
    
    expect(wrapper.find('#jestConfidence').exists()).toBeFalsy();
  });

  it('should render consistently', () => {
    const wrapper = getLabel();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render consistently for isClickable not enabled', () => {
    const wrapper = getLabel({
      ...initialProps,
      isClickable: false,
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render consistently for being selected', () => {
    const wrapper = getLabel({
      ...initialProps,
      selected: true,
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});