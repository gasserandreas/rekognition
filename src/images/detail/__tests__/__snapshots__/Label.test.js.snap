// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faces test suite should render consistently 1`] = `
.c0 {
  border: 1.5px solid #6B778C;
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block;
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  color: #0052CC;
  border-color: #0052CC;
}

.c0 span {
  font-size: 0.75rem;
}

.c0:hover {
  cursor: pointer;
  background-color: #0052CC;
  color: #ffffff;
}

<Label
  isClickable={true}
  label={
    Object {
      "confidence": 100,
      "id": "f7ba4488-4db3-484d-9af9-37f1a7a2aeb2",
      "instances": Array [
        Object {
          "height": 100,
          "left": 80,
          "top": 70,
          "width": 90,
        },
      ],
      "name": "name",
      "parents": Array [],
      "value": "value",
    }
  }
  onClick={[MockFunction]}
  selected={false}
>
  <styled.div
    isClickable={true}
    onClick={[MockFunction]}
    selected={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  border: 1.5px solid ",
              "#6B778C",
              ";
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;

  span {
    font-size: 0.75rem;
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isClickable={true}
      onClick={[MockFunction]}
      selected={false}
    >
      <div
        className="c0"
        onClick={[MockFunction]}
        selected={false}
      >
        name
         
        <span
          id="jestConfidence"
        >
          (
          100.00
          )
        </span>
      </div>
    </StyledComponent>
  </styled.div>
</Label>
`;

exports[`Faces test suite should render consistently for being selected 1`] = `
.c0 {
  border: 1.5px solid #6B778C;
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block;
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  cursor: pointer;
  background-color: #0052CC;
  color: #ffffff;
}

.c0 span {
  font-size: 0.75rem;
}

<Label
  isClickable={true}
  label={
    Object {
      "confidence": 100,
      "id": "f7ba4488-4db3-484d-9af9-37f1a7a2aeb2",
      "instances": Array [
        Object {
          "height": 100,
          "left": 80,
          "top": 70,
          "width": 90,
        },
      ],
      "name": "name",
      "parents": Array [],
      "value": "value",
    }
  }
  onClick={[MockFunction]}
  selected={true}
>
  <styled.div
    isClickable={true}
    onClick={[MockFunction]}
    selected={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  border: 1.5px solid ",
              "#6B778C",
              ";
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;

  span {
    font-size: 0.75rem;
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isClickable={true}
      onClick={[MockFunction]}
      selected={true}
    >
      <div
        className="c0"
        onClick={[MockFunction]}
        selected={true}
      >
        name
         
        <span
          id="jestConfidence"
        >
          (
          100.00
          )
        </span>
      </div>
    </StyledComponent>
  </styled.div>
</Label>
`;

exports[`Faces test suite should render consistently for isClickable not enabled 1`] = `
.c0 {
  border: 1.5px solid #6B778C;
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block;
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
}

.c0 span {
  font-size: 0.75rem;
}

<Label
  isClickable={false}
  label={
    Object {
      "confidence": 100,
      "id": "f7ba4488-4db3-484d-9af9-37f1a7a2aeb2",
      "instances": Array [
        Object {
          "height": 100,
          "left": 80,
          "top": 70,
          "width": 90,
        },
      ],
      "name": "name",
      "parents": Array [],
      "value": "value",
    }
  }
  onClick={[MockFunction]}
  selected={false}
>
  <styled.div
    isClickable={false}
    onClick={[MockFunction]}
    selected={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  border: 1.5px solid ",
              "#6B778C",
              ";
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  display: inline-block
  font-size: 0.8rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;

  span {
    font-size: 0.75rem;
  }

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isClickable={false}
      onClick={[MockFunction]}
      selected={false}
    >
      <div
        className="c0"
        onClick={[MockFunction]}
        selected={false}
      >
        name
         
        <span
          id="jestConfidence"
        >
          (
          100.00
          )
        </span>
      </div>
    </StyledComponent>
  </styled.div>
</Label>
`;
